feat(java,info): refactor `EdgeInfo` creation using access-level compliant builder, a…

Jarvx200 opened the pull request:
<!--
Thanks for contributing to GraphAr.
If this is your first pull request you can find detailed information on [CONTRIBUTING.md](https://github.com/apache/graphar/blob/main/CONTRIBUTING.md)

The Apache GraphAr (incubating) community has restrictions on the naming of pr title. You can find instructions in
[CONTRIBUTING.md](https://github.com/apache/graphar/blob/main/CONTRIBUTING.md#title) too.
-->

### Reason for this PR

Boiler plate heavy telescoping constructor inside `EdgeInfo` #740 
<!-- 
Why are you proposing this change? If this is already tracked in an issue, please link to the issue here.
Explaining clearly why this change is beneficial is important for the reviewers to understand the context.
-->


### What changes are included in this PR?

Refactor of telescoping constructor to builder pattern while keeping API access levels and compatibility.

### Are these changes tested?

Yes, using the new builder in `TestUtil` and running the dependent `GraphSaverTest`

### Are there any user-facing changes?

No

Jarvx200 commented:
If the pattern is correct, I can continue by deprecating the old constructors and switching to the new method

yangxk1 reviewed (commented):
To be honest, [`Lombok`](https://projectlombok.org/) is very suitable for adding builder, but it will bring more dependencies

Jarvx200 commented:
Ok sure, I will start working on the first two problems, the third one would imply temporary builder fields that will eventually get transformed with null checking to triplet, but I can wait for further insights on the problem and can implement if ok.

Jarvx200 commented:
Changes:
- Added static factory to builder as requested 
- Added methods for list types to get appended (Note: since `PropertyGroups` is immutable reducing was needed to get the last instance with all the appended groups)
- Added support to construct `EdgeTriplet` with variables inside builder

yangxk1 reviewed (commented):
The overall code is good, I proposed some small issues, and it will be better after solving!

Jarvx200 commented:
Hello, I went through all the proposed issues and provided fixes and testing!

yangxk1 commented on maven-projects/info/src/main/java/org/apache/graphar/info/EdgeInfo.java:
```diff
@@ -51,6 +53,192 @@ public class EdgeInfo {
+            return this;
+        }
+
+        private EdgeInfoBuilder addPropertyGroups(PropertyGroups propertyGroups) {
```
change the access modifiers and name to `public propertyGroups`

yangxk1 commented on maven-projects/info/src/main/java/org/apache/graphar/info/EdgeInfo.java:
```diff
@@ -51,6 +53,192 @@ public class EdgeInfo {
+            return this;
+        }
+
+        private EdgeInfoBuilder adjacentLists(Map<AdjListType, AdjacentList> adjacentLists) {
```
Also public

yangxk1 reviewed (commented):
Hi @Jarvx200, Your code is already very good, but needs to format and add license to pass ci test. And I found that there is some  mistake about access modifiers and name. 
It would be better if you could add more tests (e.g., for addPropertyGroup` and `addAdjacentlists` )

Jarvx200 commented:
Hello! I will start working on those problems, write some more tests to make sure everything is compliant.

Jarvx200 commented:
Hello, I fixed the issues above and added a missing `final` to `TestUtil`. If there is any further testing required please tell me.
